scripts/install_R_source.sh

```diff
@@ -157,7 +158,9 @@ rm -rf "R.tar.gz"
 cp /usr/bin/checkbashisms /usr/local/bin/checkbashisms
 
 # shellcheck disable=SC2086
-apt-get remove --purge -y ${BUILDDEPS}
+if [ "${PURGE_BUILDDEPS}" == "TRUE" ]; then
```

Comment:
@eitsupi you said:

> it would be better to use the same script with the flag added to install_R_source.sh and skipped the "build R" steps so that we don't have to have almost identical copies in two files.

I am adding a flag to install_R_source.sh right here though.  what two files are you referring two as almost identical copies?

Reply:
Perhaps the assumptions were not engaged. I had imagined something like the following:

```dockerfile
FROM rocker/r-ver:4.3.2 as builder

FROM FROM nvidia/cuda:11.8.0-cudnn8-devel-ubuntu22.04
COPY --from=builder /foo/bar
COPY scripts/install_R_source.sh /rocker_scripts/install_R_source.sh
ENV SKIP_BUILD_R="true"
RUN /rocker_scripts/install_R_source.sh
```

Reply:
Thanks, I see what you mean.  That looks rather more complicated though. 
-  It requires using the multi-stage build, and i'm not sure how easy that is to add to our existing build stack.   
- Copying `/foo/bar` isn't so clean, because our standard build installs R_HOME into /usr/local, with binaries in /usr/local/bin, etc, rather than freestanding.  Yes, I proposed a solution around this in my previous proposal, but it is a significant change that could break some user expectations.
- SKIP_BUILD_R would have to wrap around thin install-time deps,  building of R and and clean up and purge of dependencies.  That's fine, but it feels more natural to me to have an option that just avoids the `apt remove` step.  After all, given that `apt remove` has unexpected consequences, it seems reasonable to be able to opt out of that without opting out of the rest of it.  and intuitively, it's easier I think to guess what `install_R_source.sh` does behaves in response to a option of `PURGE_BUILDDEPS=FALSE` than it is to guess what `install_R_source.sh` does in response to an option of `SKIP_BUILD_R`.

What's the downside of the approach here? I still don't see what's been duplicated.